yaptide/application.py

```diff
@@ -51,5 +51,27 @@ def create_app():
     return app
 
 
+def find_git_executable():
+    """Function returns git executable path or None"""
+    git_path = shutil.which('git')
+    return git_path
+
+
+def check_submodules():
```

Comment:
I've tried this approach in following way:

1. Cloned repo without submodules (on Windows):

```powershell
PS C:\Users\Legion\workspace\1\yaptide> git clone https://github.com/yaptide/yaptide.git
```

and checkout branch `770-need-a-warning-if-repository-was-clone-without-submodules`

2. Started yaptide with the method described in README.md:

```
PS C:\Users\Legion\workspace\1\yaptide> docker compose up -d --build       
```

I've got an error as expected:

```
[+] Running 3/3
 ✔ Container yaptide_redis       Healthy                                                                           0.0s
 ✔ Container yaptide_postgresql  Healthy                                                                           0.0s
 ✘ Container yaptide_flask       Error                                                                             0.0s
dependency failed to start: container yaptide_flask exited (1)
```

But the error message didn't said much:

```python
2024-03-10 17:09:11 Traceback (most recent call last):
2024-03-10 17:09:11   File "/usr/local/bin/flask", line 8, in <module>
2024-03-10 17:09:11     sys.exit(main())
2024-03-10 17:09:11   File "/usr/local/lib/python3.10/site-packages/flask/cli.py", line 1064, in main
2024-03-10 17:09:11     cli.main()
2024-03-10 17:09:11   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1078, in main
2024-03-10 17:09:11     rv = self.invoke(ctx)
2024-03-10 17:09:11   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1688, in invoke
2024-03-10 17:09:11     return _process_result(sub_ctx.command.invoke(sub_ctx))
2024-03-10 17:09:11   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
2024-03-10 17:09:11     return ctx.invoke(self.callback, **ctx.params)
2024-03-10 17:09:11   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 783, in invoke
2024-03-10 17:09:11     return __callback(*args, **kwargs)
2024-03-10 17:09:11   File "/usr/local/lib/python3.10/site-packages/click/decorators.py", line 92, in new_func
2024-03-10 17:09:11     return ctx.invoke(f, obj, *args, **kwargs)
2024-03-10 17:09:11   File "/usr/local/lib/python3.10/site-packages/click/core.py", line 783, in invoke
2024-03-10 17:09:11     return __callback(*args, **kwargs)
2024-03-10 17:09:11   File "/usr/local/lib/python3.10/site-packages/flask/cli.py", line 912, in run_command
2024-03-10 17:09:11     raise e from None
2024-03-10 17:09:11   File "/usr/local/lib/python3.10/site-packages/flask/cli.py", line 898, in run_command
2024-03-10 17:09:11     app = info.load_app()
2024-03-10 17:09:11   File "/usr/local/lib/python3.10/site-packages/flask/cli.py", line 309, in load_app
2024-03-10 17:09:11     app = locate_app(import_name, name)
2024-03-10 17:09:11   File "/usr/local/lib/python3.10/site-packages/flask/cli.py", line 236, in locate_app
2024-03-10 17:09:11     return find_best_app(module)
2024-03-10 17:09:11   File "/usr/local/lib/python3.10/site-packages/flask/cli.py", line 64, in find_best_app
2024-03-10 17:09:11     app = app_factory()
2024-03-10 17:09:11   File "/usr/local/app/yaptide/application.py", line 20, in create_app
2024-03-10 17:09:11     check_submodules()
2024-03-10 17:09:11   File "/usr/local/app/yaptide/application.py", line 66, in check_submodules
2024-03-10 17:09:11     raise RuntimeError("Git executable not found.")
2024-03-10 17:09:11 RuntimeError: Git executable not found.
```

It seems the approach with git is OK for developer-mode, but if a user would like to deploy the app, it won't help.

Reply:
Obviously there is no `git` binary in docker container used for deployment.

Reply:
In comment given here: https://github.com/yaptide/yaptide/pull/875#discussion_r1512805385 I've asked about pros and cons. It seems that is first `cons`. Better to consider `pros` and `cons` before starting implementation :)

Reply:
As for another proposition - lets stick to git as it seems quite flexible solution. 
I think that `check_submodule` could be moved into standalone python file, located in say `yaptide/admin` package.
Then you could call it from `Dockerfile-flask` or other to ensure that submodules were cloned.
The logic can stay as it is.